tools/labs/templates/deferred_work/3-4-5-deferred/kernel/deferred.c

```diff
@@ -226,7 +226,7 @@ static int deferred_init(void)
 	cdev_add(&dev.cdev, MKDEV(MY_MAJOR, MY_MINOR), 1);
 
 	/* TODO 1: Initialize timer. */
-	setup_timer(&dev.timer, timer_handler, (unsigned long) &dev);
+	timer_setup(&dev.timer, timer_handler, 0);
```

Comment:
Not related to the API changes, but I think it would be better if we would use &my_data->timer instead of dev.timer. Maybe add a new commit on top to do this.